docs/source/en/generation_strategies.mdx

```diff
@@ -0,0 +1,287 @@
+producing highly repetitive results.
+
+## Customize text generation
+
+You can override any `generation_config` by passing the parameters and their values directly to the generate() method,
+e.g. `my_model.generate(inputs, num_beams=4, do_sample=True)`.
+
+Even if the default decoding strategy mostly works for your task, you can still tweak a few things. Some of the
+commonly adjusted parameters include:
+
+- `max_new_tokens`: the maximum number of tokens to generate. In other words, the size of the output sequence, not
+including the tokens in the prompt.
+- `max_length`: similar to `max_new_tokens`, this parameter limits the size of the generated output, however, it takes
+into account the input's size.
+- `num_beams`: by specifying a number of beams higher than 1, you are effectively switching from greedy search to
+beam search. This strategy evaluates several hypotheses at each time step and eventually chooses the hypothesis that
+has the overall highest probability for the entire sequence. This has the advantage of identifying high-probability
+sequences that start with a lower probability initial tokens and would've been ignored by the greedy search.
+- `do_sample`: If set to `True`, this parameter enables such decoding strategies as multinomial sampling, beam-search
+multinomial sampling, Top-K sampling and Top-p sampling. All these strategies select the next token from the probability
+distribution over the entire vocabulary with various strategy-specific adjustments.
+
+## Decoding strategies
+
+Certain combinations of the `generate()` parameters, and ultimately `generation_config`, can be used to enable specific
+decoding strategies. If you are new to this concept, we recommend reading [this blog post that illustrates how common decoding strategies work](https://huggingface.co/blog/how-to-generate).
+
+Here, we'll show what parameters control some of the strategies, and illustrate how you can use them.
+
+### Greedy Search
+Greedy search is the default decoding strategy. For generate() method to use it, `num_beams` has to be set to 1, and
+`do_sample=False`. Because it is a default strategy, you do not have to pass any parameters to `generate()` method to enable it.
+
+```python
+>>> from transformers import AutoModelForCausalLM, AutoTokenizer
+
+>>> prompt = "I look forward to"
+>>> checkpoint = "distilgpt2"
+
+>>> tokenizer = AutoTokenizer.from_pretrained(checkpoint)
+>>> input_ids = tokenizer(prompt, return_tensors="pt").input_ids
+
+>>> model = AutoModelForCausalLM.from_pretrained(checkpoint)
+>>> outputs = model.generate(input_ids)
+>>> tokenizer.batch_decode(outputs, skip_special_tokens=True)
+['I look forward to seeing you all again!\n\n\n\n\n\n\n\n\n\n\n']
+```
+
+### Contrastive Search
+The Contrastive search decoding strategy was proposed in the 2022 paper <a href="https://arxiv.org/abs/2202.06417">"A Contrastive Framework for Neural Text Generation"</a>.
```

Comment:
I initially used markdown, but it wasn't rendering for some reason. Maybe the quotation marks were the issue. I'll switch back to markdown, and hopefully, it will render.